pkg/agent/flowexporter/connections/conntrack_test.go

```diff
@@ -76,28 +82,59 @@ func TestConnTrackSystem_DumpFlows(t *testing.T) {
 	}
 	// Test the DumpFlows implementation of connTrackSystem
 	mockNetlinkCT := connectionstest.NewMockNetFilterConnTrack(ctrl)
-	connDumperDPSystem := &connTrackSystem{
-		nodeConfig,
-		serviceCIDR,
-		mockNetlinkCT,
-	}
+	connDumperDPSystem, acctErr, timestampErr := NewConnTrackSystem(nodeConfig, serviceCIDR)
+	connDumperDPSystem.connTrack = mockNetlinkCT
 	// Set expects for mocks
 	mockNetlinkCT.EXPECT().Dial().Return(nil)
 	mockNetlinkCT.EXPECT().DumpFilter(conntrack.Filter{}).Return(testFlows, nil)
 
-	conns, err := connDumperDPSystem.DumpFlows(openflow.CtZone)
+	conns, conntrackMetrics, err := connDumperDPSystem.DumpFlows(openflow.CtZone)
 	if err != nil {
 		t.Errorf("Dump flows function returned error: %v", err)
 	}
 	assert.Equal(t, 1, len(conns), "number of filtered connections should be equal")
+
+	if acctErr == nil {
+		conntrackAcct, err := sysctl.GetSysctlNet("netfilter/nf_conntrack_acct")
+		// Ignore the permission denied error if test cluster doesn't have privilege
+		if err != nil && !os.IsPermission(err) {
+			t.Errorf("error when getting nf_conntrack_acct: %v", err)
+		} else if err == nil {
+			assert.Equal(t, 1, conntrackAcct, "net.netfilter.nf_conntrack_acct value should be 1")
+		}
+	}
+
+	if timestampErr == nil {
+		conntrackTimestamping, err := sysctl.GetSysctlNet("netfilter/nf_conntrack_timestamp")
+		// Ignore the permission denied error if test cluster doesn't have privilege
+		if err != nil && !os.IsPermission(err) {
+			t.Errorf("error when getting nf_conntrack_timestamp: %v", err)
+		} else if err == nil {
+			assert.Equal(t, 1, conntrackTimestamping, "net.netfilter.nf_conntrack_timestamp value should be 1")
+		}
+	}
+
+	maxConns, err := sysctl.GetSysctlNet("nf_conntrack_max")
+	// Ignore the permission denied error if test cluster doesn't have privilege
```

Comment:
The comment is misleading: these tests are not run in clusters. The Go unit tests are run on Linux & Windows VMs (well these ones only on Linux because of the build tag). Unit tests should ideally never require special privileges. So right now in CI, we will never be able to exercise the normal code path, because the sysctl stuff will always fail.

Reply:
Thanks, I will modify the comment. By the way, the Github CI running unit tests didn't fail on the sysctl, so this if condition is used to bypass the asset statement if the sysctl command failed because of the permission error on other CI platforms.

Reply:
I thought that these checks were introduced because a CI job was failing originally?

BTW my earlier comment is wrong: it seems we only run unit tests on Linux, not macOS

Reply:
Yes. Actually, the NewConnTrackSystem function meets the permission error when setting nf_conntrack_acct and nf_conntrack_timestamp and these error logs have been there before. Since we add assert statements for these two values in this unit test, the CI job begins to fail because test VM could get these values using sysctl command so there is a mismatch. In the lastest commit, we fix this failure by adding the error return value of NewConnTrackSystem, if the setting process failed we will not do the assert statement. But since I'm not sure if the GetSysctlNet will fail or not when setting process succeeded, so I add the if condition for GetSysctlNet too.